contracts/Vault.sol

```diff
@@ -148,18 +147,30 @@ contract Vault is ExchangeRate, OwnableUpgradeSafe, PausableUpgradeSafe {
         require(aTokenAmount >= depositorToPrincipal[msg.sender], "atoken must be more than depositorPrincipal");
         require(depositorReserve >= aTokenAmount.sub(depositorToPrincipal[msg.sender]), "atoken must be more than depositorPrincipal");
         depositorReserve = depositorReserve.sub(
-            aTokenAmount.sub(depositorToPrincipal[msg.sender]) 
+            aTokenAmount.sub(depositorToPrincipal[msg.sender])
         );
         require(principal >= depositorToPrincipal[msg.sender], "principal must be over depositorPrincipal");
         principal = principal.sub(depositorToPrincipal[msg.sender]);
         depositorToPrincipal[msg.sender] = 0;
         // withdraw atokens
         aToken.approve(address(aavePool), aTokenAmount);
-        uint256 withdrawn = aavePool.withdraw(
-            address(token),
-            aTokenAmount,
-            msg.sender
-        );
+        uint256 aTokenBalance = aToken.balanceOf(address(this));
+
+        // @dev in an edge case where the btokenToToken() math is imprecise,
+        // @dev aTokenAmount could be higher than total aToken balance when the withdraw is owned by single owner
+        if (aTokenAmount >= aTokenBalance) {
```

Comment:
per our chat, in the interest of breaking up the logic and into smaller logical and testable pieces i might break this up well even though it looks trivial. make a private function `_aavePoolWithdraw(amount)` that would be called with different amounts in each case.